experiment/run_experiment.py

```diff
@@ -59,36 +59,54 @@
 _OSS_FUZZ_CORPUS_BACKUP_URL_FORMAT = (
     'gs://{project}-backup.clusterfuzz-external.appspot.com/corpus/'
     'libFuzzer/{fuzz_target}/public.zip')
+DEFAULT_MAX_CONCURRENT_BUILDS = 30
 
 
+def set_default_config_values(config: Dict):
+    """Set the default configuration values if they are not specified."""
+    config['concurrent_builds'] = config.get('concurrent_builds',
+                                             DEFAULT_MAX_CONCURRENT_BUILDS)
+
+
+# pylint: disable=too-many-locals, too-many-statements, too-many-branches
 def read_and_validate_experiment_config(config_filename: str) -> Dict:
     """Reads |config_filename|, validates it, finds as many errors as possible,
     and returns it."""
     config = yaml_utils.read(config_filename)
+    set_default_config_values(config)
     filestore_params = {'experiment_filestore', 'report_filestore'}
     cloud_config = {'cloud_compute_zone', 'cloud_project'}
     docker_config = {'docker_registry'}
-    string_params = cloud_config.union(filestore_params).union(docker_config)
-    int_params = {'trials', 'max_total_time'}
+    worker_config = {'worker_pool_name'}
+    string_params = cloud_config.union(filestore_params).union(
+        docker_config).union(worker_config)
+    int_params = {'trials', 'max_total_time', 'concurrent_builds'}
+    optional_params = worker_config
     required_params = int_params.union(filestore_params).union(docker_config)
     bool_params = {'private', 'merge_with_nonprivate'}
 
     local_experiment = config.get('local_experiment', False)
     snapshot_period = config.get('snapshot_period',
                                  experiment_utils.DEFAULT_SNAPSHOT_SECONDS)
-    if not local_experiment:
+    if local_experiment:
+        optional_params = optional_params.union(cloud_config)
+    else:
         required_params = required_params.union(cloud_config)
 
     valid = True
     if 'cloud_experiment_bucket' in config or 'cloud_web_bucket' in config:
         logs.error('"cloud_experiment_bucket" and "cloud_web_bucket" are now '
                    '"experiment_filestore" and "report_filestore".')
-
-    for param in required_params:
-        if param not in config:
+    missing_params = required_params - config.keys()
```

Comment:
Do two loops instead of this.
Go through missing_params and set valid = False and log
then go through optional_params - config.keys() and log that hey are missing.

Reply:
Done, I also split the function to make it look better.